### PR TITLE
Doc changes for 3.2 comparing with 3.1 branches

### PR DESCRIPTION
**Pull request by xuanyuanking**
For review the 3.2 only changes.

### CODE REVIEW

File: docs/sql-ref-ansi-compliance.md [hunk 1]
```diff
@@ -176,14 +273,20 @@ The behavior of some SQL functions can be different under ANSI mode (`spark.sql.
...
+  - `next_day`: This function throws `IllegalArgumentException` if input is not a valid day of week.
 
 ### SQL Operators
 
 The behavior of some SQL operators can be different under ANSI mode (`spark.sql.ansi.enabled=true`).
   - `array_col[index]`: This operator throws `ArrayIndexOutOfBoundsException` if using invalid indices.
   - `map_col[key]`: This operator throws `NoSuchElementException` if key does not exist in map.
-  - `CAST(string_col AS TIMESTAMP)`: This operator should fail with an exception if the input string can't be parsed.
-  - `CAST(string_col AS DATE)`: This operator should fail with an exception if the input string can't be parsed.
+
```
Comment (srielau) [inline docs/sql-ref-ansi-compliance.md:283]: @cloud-fan Odd, why was this removed? It still holds in my DBR 10 build (and it seems right)

File: docs/sql-ref-syntax-qry-select-transform.md [hunk 2]
```diff
@@ -24,6 +24,15 @@ license: |
 The `TRANSFORM` clause is used to specify a Hive-style transform query specification
```
Comment (srielau) [inline docs/sql-ref-syntax-qry-select-transform.md:24]: We don't have this file in DBR. I take it we don't support TRANSFORM?

File: docs/sql-ref-syntax.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ Spark SQL is Apache Spark's module for working with structured data. The SQL Syn
```
Comment (srielau) [inline docs/sql-ref-syntax.md:23]: This maps to our index.md which already has language. leave it alone.

File: docs/sql-ref-ansi-compliance.md (hunk 1 above)
Comment (xuanyuanking) [reply docs/sql-ref-ansi-compliance.md:283]: Seems we used the wrong commits here, the latest file changes for the `sql-ref-ansi-compliance.md` should be:
image

File: docs/sql-ref-syntax-qry-select-transform.md (hunk 2 above)
Comment (xuanyuanking) [reply docs/sql-ref-syntax-qry-select-transform.md:24]: Just confirmed with @gatorsmile, we support `TRANSFORM` but do not encourage the customer to use it.

File: docs/sql-ref-ansi-compliance.md (hunk 1 above)
Comment (cloud-fan) [reply docs/sql-ref-ansi-compliance.md:283]: We decided to allow "group by alias" in ANSI mode. Are you sure it's still forbidden in 10.0? cc @gengliangwang
